src/app/(sidebar)/transaction/build/components/TransactionXdr.tsx

```diff
@@ -121,6 +129,39 @@ export const TransactionXdr = () => {
         }, [] as any[]);
       };
 
+      const flagTotal = (val: string[], operations: OptionFlag[]) => {
+        const total = optionsFlagDetails(operations, val).total;
+
+        return total > 0 ? BigInt(total) : null;
+      };
+
+      const formatSignerValue = (val: OptionSigner | undefined) => {
+        if (!val) {
+          return null;
+        }
+
+        const weight = val?.weight ? BigInt(val.weight) : "";
+        let key: string | any = "";
+
+        switch (val.type) {
+          case "ed25519PublicKey":
+            key = val.key || "";
+            break;
+          case "sha256Hash":
+            // TODO: format value
+            key = val.key;
+            break;
+          case "preAuthTx":
+            // TODO: format value
+            key = val.key;
+            break;
```

Comment:
I couldn't figure out how to decode/encode these values for XDR JSON; will try to fix these a bit later.

Reply:
Fixed